community/cypher/cypher/src/test/scala/org/neo4j/cypher/CypherTypeAcceptanceTest.scala

```diff
@@ -70,4 +70,9 @@ class CypherTypeAcceptanceTest extends ExecutionEngineFunSuite {
 
     result should be(empty)
   }
+
+  test("should be consistent in failing either statically or at runtime when trying to access an array with a non-integer index") {
+    a [SyntaxException] should be thrownBy execute("WITH [1,2,3,4,5] AS array, 3.14 AS idx RETURN array[idx]")
```

Comment:
All semantic errors are mapped to `SyntaxException`s. Not that there is no `SemanticError` public exception. We might want to change the current situation, but this comment is unrelated to this specific fix.